src/Heatmap/HeatMapInterpolationPoints.swift

```diff
@@ -0,0 +1,399 @@
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+import Foundation
+import GooglePlaces
+import GoogleMaps
+import GoogleMapsUtils
+
+/// This class will create artificial points in surrounding locations with appropriate intensities interpolated by neighboring intensity values.
+class HeatMapInterpolationPoints {
+    
+    /// The input data set; each entry contains the coordinates and the 
+    private var data = [[Double]]()
+    
+    /// The heat map layer
+    private let heatMapLayer: GMUHeatmapTileLayer = GMUHeatmapTileLayer()
+    
+    /// The heat map colors and gradient
+    private var gradientColors = [UIColor.green, UIColor.red]
+    private var gradientStartPoints = [0.005, 0.7] as [NSNumber]
+    
+    /// The user should be able to access thisdata as well as the size of the data
+    public var heatMapPoints = [GMUWeightedLatLng]()
+    
+    // MARK: Functions that parse given data needed to build an interpolated heat map from
+    
+    /// Takes in the data set file and calculates the weights of the given points; these values are used for interpolation later
+    ///
+    /// - Parameter file: The input data set file; it must have a json extension.
+    public func setData(file: String) {
+        data.removeAll()
+        do {
+            guard let path = Bundle.main.url(forResource: file, withExtension: "json") else {
+                print("Data set path error")
+                return
+            }
+            let data = try Data(contentsOf: path)
+            let json = try JSONSerialization.jsonObject(with: data, options: [])
+            guard let object = json as? [[String: Any]] else {
+                print("Could not read the JSON file or file is empty")
+                return
+            }
+            for item in object {
+                let lat: Double = item["lat"] as? CLLocationDegrees ?? 0.0
+                let lng: Double = item["lng"] as? CLLocationDegrees ?? 0.0
+                append(lat: lat, long: lng)
+            }
+        } catch {
+            print(error.localizedDescription)
+        }
+    }
+    
+    public func addPoints(pointList: [[Double]]) {
+        for point in pointList {
+            append(lat: point[0], long: point[1])
+        }
+    }
+    
+    /// A helper function that adds a given set of coordinates to the list of points or increments an existing coordinate
+    ///
+    /// - Parameters:
+    ///   - lat: The latitude value of the new point.
+    ///   - long: The longitude value of the new point.
+    private func append(lat: Double, long: Double) {
```

Comment:
This currently assumes that weight on a given lat/lng is provided based on repitition of that same lat/lng in the list. I'm not convinced this is a common use case. Rather, input will likely be provided as a lat,lng with a given weight already. 

Taking the example [here](https://mgimond.github.io/Spatial/spatial-interpolation.html), you can imagine an input where you have a value (e.g. a temperature reading) at a given lat lng.

Given this, consider having `addWeightedLatLngs(latLngs: [GMUWeightedLatLng])` and `addWeightedLatLng(latLng: GMUWeightedLatLng)` as an API instead.

Reply:
Sounds good! I'll work on that ASAP